posts/hello-world-in-kafka-using-python.md

```diff
@@ -19,30 +19,36 @@ You might have got a hint of it already. This blog is about understanding what i
 ## What is Kafka? Why should one use it?
 In short, Kafka is a distributed streaming and message queueing platform.
 
-> Oh wait! What does that even mean? How different is it from traditional databases? 
+> Oh wait! What does that even mean?
 
+A streaming platform is a system that can perform the following:
+1. Store and process streams of data in real time
+2. Allow applications and/or services to publish and subscribe to these data streams at scale
 
-Simplified explanation of what are streaming and messaging queues and how they are different from traiditional databases.
+> Interesting! How different or similar is it from traditional databases?
 
+Kafka is NOT a database. Nevertheless, it has few similarities and quite an amount of differences with respect to traditional databases.
 
-> That sounds convincing! But why would one need a system like this?
+Like traditional databases, you can store data into Kafka which can be persistent, checksummed and replicated for fault tolerance. This is one reason why Kafka is not just a message queue but a streaming platform. In addition, Kafka has something called as topics which is similar to tables in PostgreSQL or collections in MongoDB. 
 
+Unlike traditional databases, Kafka deals with real time data that can be published or consumed by multiple applications at the same time. Alongside, Kafka allows you to process the data streams but this is totally different when compared to performing CRUD operations or running simple to complex queries on traditional databases. 
```

Comment:
Database too can store real time data, you need to give an example about publish and subscribe with an image. Add a question, "What does publish and subscribe do? How is it helpful?